dlt/destinations/dataset/ibis_relation.py

```diff
@@ -32,30 +33,13 @@ def __init__(
         super().__init__(readable_dataset=readable_dataset)
         self._ibis_object = ibis_object
 
-    def _query(self) -> Any:
-        from dlt.helpers.ibis import ibis
-
-        ibis_expr = self._ibis_object
-        caps = self._dataset.sql_client.capabilities
-        target_dialect = caps.sqlglot_dialect
-
-        # render sql directly if possible
-        # NOTE: ibis is optimized for reading a real schema from db and pushing back optimized sql
-        #  - it quotes all identifiers, there's no option to get unqoted query
-        #  - it converts full lists of column names back into star
-        #  - it optimizes the query inside, adds meaningless aliases to all tables etc.
-        if target_dialect not in TRANSPILE_VIA_DEFAULT:
-            if target_dialect == "tsql":
-                # NOTE: Ibis uses the product name "mssql" as the dialect instead of the official "tsql".
-                return ibis.to_sql(ibis_expr, dialect="mssql")
-            else:
-                return ibis.to_sql(ibis_expr, dialect=target_dialect)
-
-        # here we need to transpile to ibis default and transpile back to target with sqlglot
-        # NOTE: ibis defaults to the duckdb dialect, if a dialect is not passed
-        sql = ibis.to_sql(ibis_expr)
-        sql = sqlglot.transpile(sql, write=target_dialect)[0]
-        return sql
+    def _query(self) -> sge.Query:
+        from dlt.helpers.ibis import duckdb_compiler
+
+        select_query = duckdb_compiler.to_sqlglot(self._ibis_object)
```

Comment:
For converting an ibis expression into sqlglot, you always need a compiler which is destination dependent. I think the compiler will fail if you use function that are known not to exist on a specific destination, and certain materializations will also occur in this step. If you have an expression that counts rows, the alias for the result will be created here, which will be `CountStar(*)` for duckdb, but which is not a valid identifier for bigquery for example (see comment in query normalizer)